hs_core/views/utils.py

```diff
@@ -1070,90 +1071,101 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
 
     unzip_path = None
     try:
-        # unzip to a temporary folder
-        unzip_path = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
-        dir_file_list = istorage.listdir(unzip_path)
-        unzip_subdir_list = dir_file_list[0]
-        # list all files to be moved into the resource
-        unzipped_files = listfiles_recursively(istorage, unzip_path)
-        unzipped_foldername = os.path.basename(unzip_path)
-        override_tgt_paths = []
-        for sub_dir_name in unzip_subdir_list:
-            dest_sub_path = os.path.join(os.path.dirname(zip_with_full_path), sub_dir_name)
-            if istorage.exists(dest_sub_path):
-                override_tgt_paths.append(dest_sub_path)
-
-        res_files = []
-        for unzipped_file in unzipped_files:
-            res_files.append(IrodsFile(unzipped_file, istorage))
-        meta_files = []
-        if ingest_metadata:
-            res_files, meta_files, map_files = identify_metadata_files(res_files)
-
-        for file in res_files:
-            destination_file = _get_destination_filename(file.name, unzipped_foldername)
-            if istorage.exists(destination_file):
-                override_tgt_paths.append(destination_file)
-
-        if not overwrite and override_tgt_paths:
-            override_simplified_paths = [tgt_path.split('data/contents/')[1] for tgt_path in override_tgt_paths]
-            raise FileOverrideException('move would overwrite {}'.format(', '.join(override_simplified_paths)))
-
-        for override_tgt_path in override_tgt_paths:
-            if istorage.exists(override_tgt_path):
-                if resource.resource_type == "CompositeResource":
-                    aggregation_object = resource.get_file_aggregation_object(override_tgt_path)
-                    if aggregation_object:
-                        aggregation_object.logical_delete(user)
+        if unzip_to_folder:
```

Comment:
In my testing locally, if the directory already exists, this creates a double-nested directory and appends "-1" see this video:

https://user-images.githubusercontent.com/17934193/191297379-499fa688-2105-41e6-8caf-73c6239a8cde.mov

If the directory doesn't already exist, "unzip here" still creates a directory (I would expect that no directory would be created and the unzipped files would end up in the resource content root):

https://user-images.githubusercontent.com/17934193/191300571-6ee46dd7-d206-4639-b66c-b3473d65438b.mov